test/SparqlAntlrParserTestHelpers.h

```diff
@@ -1190,4 +1191,143 @@ inline auto ExistsFilter =
 
 }  // namespace matchers
 
+namespace {
+using namespace sparqlParserHelpers;
+namespace m = matchers;
+using Parser = SparqlAutomaticParser;
+using namespace std::literals;
+using Var = Variable;
+
+const ad_utility::HashMap<std::string, std::string> defaultPrefixMap{
+    {std::string{QLEVER_INTERNAL_PREFIX_NAME},
+     std::string{QLEVER_INTERNAL_PREFIX_IRI}}};
+
+template <auto F, bool testInsideConstructTemplate = false>
+auto parse =
+    [](const string& input, SparqlQleverVisitor::PrefixMap prefixes = {},
+       std::optional<ParsedQuery::DatasetClauses> clauses = std::nullopt,
+       SparqlQleverVisitor::DisableSomeChecksOnlyForTesting disableSomeChecks =
+           SparqlQleverVisitor::DisableSomeChecksOnlyForTesting::False) {
+      ParserAndVisitor p{input, std::move(prefixes), std::move(clauses),
+                         disableSomeChecks};
+      if (testInsideConstructTemplate) {
+        p.visitor_.setParseModeToInsideConstructTemplateForTesting();
+      }
+      return p.parseTypesafe(F);
+    };
+
+auto parseBlankNode = parse<&Parser::blankNode>;
+auto parseBlankNodeConstruct = parse<&Parser::blankNode, true>;
+auto parseCollection = parse<&Parser::collection>;
+auto parseCollectionConstruct = parse<&Parser::collection, true>;
+auto parseConstructTriples = parse<&Parser::constructTriples>;
+auto parseGraphNode = parse<&Parser::graphNode>;
+auto parseGraphNodeConstruct = parse<&Parser::graphNode, true>;
+auto parseObjectList = parse<&Parser::objectList>;
+auto parsePropertyList = parse<&Parser::propertyList>;
+auto parsePropertyListNotEmpty = parse<&Parser::propertyListNotEmpty>;
+auto parseSelectClause = parse<&Parser::selectClause>;
+auto parseTriplesSameSubject = parse<&Parser::triplesSameSubject>;
+auto parseTriplesSameSubjectConstruct =
+    parse<&Parser::triplesSameSubject, true>;
+auto parseVariable = parse<&Parser::var>;
+auto parseVarOrTerm = parse<&Parser::varOrTerm>;
+auto parseVerb = parse<&Parser::verb>;
```

Comment:
If you make all of them `const`, 
then probably you make sonarcloud very happy.